### DISCUSSION TITLE
Limit build trigger folders for monorepo builds

### DISCUSSION THREAD

**Discussion by lsei** [vercel/community#310] [Help] [answered]
I currently have a monorepo with 3 projects. Every time I push code to main all projects are built. The root directory for each of the projects is different. 

Is there a way to limit the builds to only trigger when files in a project's root directory have changed? 

If so, is it then possible to build all projects if files in a `common` folder are changed? (this is the current behaviour since everything builds on every push anyway)

Folder structure
```
/
├ common/ # changes should trigger all projects to rebuild/deploy
├ project1/ # changes should only build and deploy project 1
├ project2/ # changes should only build and deploy project 2
└ project3/ # changes should only build and deploy project 3
```

I've seen tangential discussions on here but haven't found this exact issue discussed.

**Comment by samcx** [comment]
@lsei You can use [Ignore Build Step](https://vercel.com/support/articles/how-do-i-use-the-ignored-build-step-field-on-vercel) to achieve this.

**Comment by anthonyshew** [comment, marked as answer]
Vercel can now skip unnecessary monorepo builds: https://vercel.com/changelog/automatically-skip-unnecessary-deployments-in-monorepos
